alleycats-tests/shared/src/test/scala/alleycats/tests/AlleycatsSuite.scala

```diff
@@ -20,7 +19,21 @@ trait AlleycatsSuite extends munit.DisciplineSuite with TestSettings with TestIn
 }
 
 sealed trait TestInstances {
-  // To be replaced by https://github.com/rickynils/scalacheck/pull/170
-  implicit def arbitraryTry[A: Arbitrary]: Arbitrary[Try[A]] =
-    Arbitrary(Gen.oneOf(arbitrary[A].map(Success(_)), arbitrary[Throwable].map(Failure(_))))
+  implicit val arbObject: Arbitrary[Object] =
+    // with some probability we select from a small set of objects
+    // otherwise make a totally new one
+    // courtesy of @johnynek
+    Arbitrary(
+      Gen.oneOf(
+        Gen.oneOf(List.fill(5)(new Object)),
+        Arbitrary.arbUnit.arbitrary.map(_ => new Object)
+      )
+    )
+
+  implicit val arbObjectF: Arbitrary[Object => Object] =
+    Arbitrary {
+      for {
+        obj <- arbitrary[Object]
+      } yield _ => obj
+    }
```

Comment:
@SimY4 this is why the antisymmetry laws pass in your tests. This creates an arbitrary function `Object => Object` that is in fact a constant (i.e., does not depend on the input). Therefore `f(a) eq f(b)` for any choice of `a` or `b` and the law seems to pass.

If you instead use a function that depends on the input e.g. `(x: Object) => Some(x)` then the law will fail.

Reply:
yes, I agree we should not use this arbitrary to test these instances. They are not lawful for the reasons mentioned in https://github.com/typelevel/cats/pull/3944/files#r677021500 so, that's a counter example.